nixos/modules/services/network-filesystems/hdfs.nix

```diff
@@ -0,0 +1,227 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let
+  cfg = config.services.hdfs;
+
+  configurationToXml = attrs:
+    let
+      xmlescape = builtins.replaceStrings ["&" "<" ">"] ["&amp;" "&lt;" "&gt;"];
+    in ''
+      <configuration>
+      ${concatStrings (mapAttrsToList (name: value: ''
+        <property>
+          <name>${xmlescape name}</name>
+          <value>${xmlescape value}</value>
+        </property>
+       '') attrs)}
+      </configuration>
+    '';
+
+  configDir = pkgs.buildEnv {
+    name = "hdfs-config-dir";
+    paths = [
+      (pkgs.writeTextDir "core-site.xml"              (configurationToXml cfg.coreSite))
+      (pkgs.writeTextDir "hdfs-site.xml"              (configurationToXml cfg.hdfsSite))
+      (pkgs.writeTextDir "log4j.properties"           cfg.logging)
+      (pkgs.writeTextDir "hadoop-metrics2.properties" cfg.metrics)
+    ];
+  };
+
+  hadoop-configured = pkgs.stdenv.mkDerivation {
+    name = "${cfg.package.name}-configured";
+    buildInputs = [ pkgs.makeWrapper ];
+    buildCommand = ''
+      mkdir -p $out/bin
+      for n in ${cfg.package}/bin/*; do
+        makeWrapper $n $out/bin/$(basename $n) \
+          --set HADOOP_CLIENT_OPTS "${concatStringsSep " " cfg.extraCmdLineOptions}" \
+          --set HADOOP_CONF_DIR    "${configDir}"
+      done
+    '';
+  };
+
+in {
+
+  options.services.hdfs = {
+    enable = mkOption {
+      description = "enable configured hadoop";
+      type = types.bool;
+      default = cfg.enable-namenode || cfg.enable-datanode || cfg.enable-balancer || cfg.enable-journalnode;
```

Comment:
Please put this in the `config` `mkMerge` list below as:

    { services.hdfs.enable = cfg.enableNamenode || cfg.enableNatanode || cfg.enableBalancer || cfg.enableJournalnode; }

This will merge nicely even though you're also using `cfg.enable` there. This prevents rebuilding the manual if one of these options is toggled.